widget/playground/src/components/TokensPanel/TokensPanel.List.tsx

```diff
@@ -0,0 +1,195 @@
+import type { TokensListProps, TokenType } from './TokensPanel.types';
+
+import {
+  Checkbox,
+  CloseIcon,
+  Divider,
+  IconButton,
+  Image,
+  ListItemButton,
+  NotFound,
+  SearchIcon,
+  Switch,
+  TextField,
+  Typography,
+  VirtualizedList,
+} from '@rango-dev/ui';
+import React, { useEffect, useLayoutEffect, useState } from 'react';
+
+import {
+  CheckList,
+  IconWrapper,
+  SelectButton,
+  SelectDeselectText,
+} from '../MultiList/MultiList.styles';
+import { InnerElementType } from '../SingleList';
+import { EmptyContainer, StyledButton } from '../SingleList/SingleList.styles';
+
+import { TokensHeaderList } from './TokensPanel.styles';
+
+const PAGE_SIZE = 20;
+const ITEM_HEIGHT = 46;
+export function TokensList(props: TokensListProps) {
+  const {
+    list,
+    onChange,
+    onChangeAll,
+    onConfirm,
+    showSelectedTokens,
+    setShowSelectedTokens,
+    isAllSelected,
+  } = props;
+  const [searchValue, setSearchValue] = useState('');
+  const [virtualList, setVirtualList] = useState(list);
+  const [hasNextPage, setHasNextPage] = useState(true);
+
+  // Filter tokens based on search criteria
+  const filteredTokens = searchValue
+    ? list.filter((token) =>
+        token.symbol.toLowerCase().includes(searchValue.toLowerCase())
+      )
+    : list;
+
+  // Tokens to display, either all or selected
+  const tokensToDisplay = showSelectedTokens
+    ? filteredTokens.filter((token) => token.checked)
+    : filteredTokens;
+
+  useEffect(() => {
+    setHasNextPage(tokensToDisplay.length > virtualList.length);
+  }, [virtualList.length]);
+
+  useLayoutEffect(() => {
+    setVirtualList(tokensToDisplay.slice(0, PAGE_SIZE));
```

Comment:
[Screencast from 23-11-01 15:52:28.webm](https://github.com/rango-exchange/rango-client/assets/120931936/674c067e-106d-4011-a369-52d1447cdef0)
Please check this

Reply:
fixed.